packages/smooth_app/lib/background/background_task_details.dart

```diff
@@ -113,12 +114,31 @@ class BackgroundTaskDetails extends AbstractBackgroundTask {
   /// Uploads the product changes.
   @override
   Future<void> upload() async {
-    // TODO(AshAman999): check returned Status
-    await OpenFoodAPIClient.saveProduct(
+    if (_product.packagings != null) {
+      // For the moment, we can only save "packagings" with V3,
+      // and V3 can only save "packagings".
+      final ProductResultV3 result =
+          await OpenFoodAPIClient.temporarySaveProductV3(
+        getUser(),
+        _product.barcode!,
+        packagings: _product.packagings,
+        language: getLanguage(),
+        country: getCountry(),
+      );
+      if (result.status != ProductResultV3.statusSuccess &&
+          result.status != ProductResultV3.statusWarning) {
+        throw Exception('Could not save product - ${result.errors}');
+      }
+      return;
```

Comment:
A bit confused about the return statement here, do you think it's needed here ?
Suppose I am offline now and I did some edits to the product name and  then to this packagings
So when I connect back to the internet, the first block ie the code from if condition would run, and then after successful, it would simply return and won't do the save query for the rest of the changes, 

Is there something I am missing

Reply:
We don't merge changes, and just save them sequentially one by one.
In the case we want to save packagings, we are going to save only this field. Therefore only in api v3, and when it's done there's nothing else to do.

Reply:
>We don't merge changes, and just save them sequentially one by one.

Thanks for the answer, now it's clear